### ISSUE TITLE
Single Line Entry Shows Multiple Lines

### ISSUE THREAD

**Issue by stuartmscott** [fyne-io/fyne#2759]
The default entry (with `Multiline` set to `false`) will still display multiple lines if set programmatically.

```go
e := widget.NewEntry()
e.SetText(`Hello
World!`)
```

![Screenshot](https://github.com/stuartmscott/fyne/raw/5fb57a536fd4e42b40f8f6cd15b98f7c4b60fc94/widget/testdata/entry/carriage_return_single_line_text.png)

https://gophers.slack.com/archives/CK5U4BU87/p1642272423064000

**Comment by matwachich** [comment]
I think the best approach to handle this is to replace all line breaks with spaces.

**Comment by andydotxyz** [comment]
I disagree, we should not mutate content that is being set, if you combine this with data binding you'll get very unexpected results.
We should either slice it or just not render incorrectly set data.
